### PR TITLE
Add peers bootstrapper integration tests

### PR DESCRIPTION
**Pull request by robskillington**
cc @xichen2020 @cw9 @martin-mao

### CODE REVIEW

File: .ci/test-integration.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+. "$(dirname $0)/variables.sh"
+
+set -e
+
+TESTS=$(go test -test.v -test.tags=integration -test.short ./integration | grep SKIP | cut -d ' ' -f 3)
```
Comment (xichen2020) [inline .ci/test-integration.sh]: :(

Do you always need the `test.` prefix? I think you can just do `go test -v -tags=integration ...`

File: client/session.go [hunk 2]
```diff
@@ -61,6 +61,8 @@ var (
 	errSessionStateNotInitial        = errors.New("session not in initial state")
 	errSessionStateNotOpen           = errors.New("session not in open state")
 	errSessionBadBlockResultFromPeer = errors.New("session fetched bad block result from peer")
+
```
Comment (xichen2020) [inline client/session.go]: nit: Remove the blank line?

File: integration/integration.go [hunk 3]
```diff
@@ -36,3 +55,147 @@ func waitUntil(fn conditionFn, timeout time.Duration) bool {
...
+	t *testing.T,
+	opts testOptions,
+	retentionOpts retention.Options,
+	newBootstrapFn storage.NewBootstrapFn,
+) *testSetup {
+	setup, err := newTestSetup(opts)
+	require.NoError(t, err)
+
+	setup.storageOpts = setup.storageOpts.SetRetentionOptions(retentionOpts)
+	setup.storageOpts = setup.storageOpts.SetNewBootstrapFn(newBootstrapFn)
```
Comment (xichen2020) [inline integration/integration.go]: Chain these?

**Comment by xichen2020** [comment]
LGTM

File: integration/disk_cleanup_test.go [hunk 4]
```diff
@@ -81,6 +81,9 @@ func waitUntilDataCleanedUp(filePathPrefix string, namespace string, shard uint3
 }
 
 func TestDiskCleanup(t *testing.T) {
+	if testing.Short() {
```
Comment (cw9) [inline integration/disk_cleanup_test.go]: curious, how to do a short run?

**Pull request merged**